compiler/hash-typecheck/src/ops/validate.rs

```diff
@@ -50,16 +50,21 @@ impl<'gs, 'ls, 'cd> Validator<'gs, 'ls, 'cd> {
 
     /// Validate the module definition of the given [ModDefId]
     pub fn validate_mod_def(&mut self, _mod_def_id: ModDefId) -> TcResult<()> {
+        // Ensure if it is a trait impl it implements all the trait members.
         todo!()
     }
 
     /// Validate the trait definition of the given [TrtDefId]
     pub fn validate_trt_def(&mut self, _trt_def_id: TrtDefId) -> TcResult<()> {
+        // Ensure Self exists?
+        // @@Design: do we allow traits without self?
         todo!()
```

Comment:
This would be a pretty weird thing to do, but is there something that prevents someone from doing it?

Reply:
Not really, it's just a question of when does Self get introduced. Like, if someone types a function as taking Self, does Self automatically get injected implicitly into the impl? Or does it always have to be explicitly stated?